packages/fetchai/skills/tac_control/handlers.py

```diff
@@ -388,21 +391,62 @@ def _handle_unidentified_dialogue(self, oef_search_msg: OefSearchMessage) -> Non
             )
         )
 
+    def _handle_success(
+        self,
+        oef_search_success_msg: OefSearchMessage,
+        oef_search_dialogue: OefSearchDialogue,
+    ) -> None:
+        """
+        Handle an oef search message.
+
+        :param oef_search_success_msg: the oef search message
+        :param oef_search_dialogue: the dialogue
+        :return: None
+        """
+        self.context.logger.info(
+            "received oef_search success message={} in dialogue={}.".format(
+                oef_search_success_msg, oef_search_dialogue
+            )
+        )
+        target_message = cast(
+            OefSearchMessage,
+            oef_search_dialogue.get_message_by_id(oef_search_success_msg.target),
+        )
+        if (
+            target_message.performative
+            == OefSearchMessage.Performative.REGISTER_SERVICE
+        ):
+            if "location" in target_message.service_description.values:
+                game = cast(Game, self.context.game)
+                game.is_registered_agent = True
```

Comment:
and upon second success set is_registered = True

Reply:
There was/is no `is_registered` variable used in the behaviour logic. 
Do you mean I introduce this as a new flag and then use it in the behaviour logic?

Reply:
Sorry, this is relevant re tac negotiation skill.